bin/metadata_generators/base_metadata_generator.py

```diff
@@ -12,4 +12,7 @@ def get_confd_metadata(self):
         return {'compilation-status': self.compilation_status}
 
     def get_file_compilation(self):
-        return [self.compilation_status, *list(self.compilation_results.values())]
+        return {
```

Comment:
Separating the data here like this is a good idea, but a dataclass would be better than a plain dict since it also carries type information and attribute access can be checked for typos more easily than keys.
Applies to all the metadata generators.

Reply:
I used ```typing.TypedDict``` because dataclasses cannot be dumped into json files

Reply:
`TypedDict` is fine too. Just for reference, dataclasses can also be converted to dicts with [`asdict()`](https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict).

Reply:
Thanks, didn't know that